File: an2_sem4/MPP/Homework/sapt56/java/service/src/main/java/ro/ubbcluj/cs/stefnmubb/festivalsellpoint/service/BiletService.java
package ro.ubbcluj.cs.stefnmubb.festivalsellpoint.service;

import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Bilet;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Spectacol;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.repo.EntityRepoException;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.repo.IBiletRepo;

public class BiletService extends AbstractService<Integer, Bilet, IBiletRepo>
    implements IBiletService {
    public BiletService(IBiletRepo repo) {
        super(repo);
    }

    @Override
    public Iterable<Bilet> getBySpectacol(Spectacol spectacol) throws EntityRepoException {
        return getRepo().getBySpectacol(spectacol);
    }
}

File: an2_sem4/MPP/Homework/sapt10/cs/FestivalSellpoint/FestivalSellpoint.Service/BiletService.cs
using FestivalSellpoint.Domain;
using FestivalSellpoint.Repo;
using System.Collections.Generic;

namespace FestivalSellpoint.Service
{
    public class BiletService : AbstractService<int, Bilet, IBiletRepo>, IBiletService
    {
        public BiletService(IBiletRepo repo) : base(repo)
        {
        }

        public IEnumerable<Bilet> GetBySpectacol(Spectacol spectacol)
            => Repo.GetBySpectacol(spectacol);
    }
}
